### ISSUE TITLE
Unable to type letters with accents in any game that runs with Proton

### ISSUE THREAD

**Issue by KairanD** [ValveSoftware/Proton#3791]
#Games affected: 
Any game that runs with Proton and requires typing. For example, The Elder Scrolls Online (306130) and Super Animal Royale (843380). The problem doesn't happen in native games, like CS:GO. So, it's a Proton/Wine issue.

#Computer configuration: 
Processor: Core i5 4670K @ 3.8GHz.
Motherboard: Gigabyte GA-Z87-D3HP-CF @ BIOS F6.
Memory: 16GB (4 x 4GB) Corsair Vengeance @ 1600MHz.
VGA: EVGA GeForce GTX 1070Ti 8GB SC Black Edition.
OS: Ubuntu 19.10 Eoan Ermine (x64).
VGA Driver: NVidia 440.59.
Proton Version: 5.0-6.

#Problem description:
I'm able to use accents normally in everything, including Steam chat: á, é, í, ó, ú, ã, ñ.

When I open any game that runs with Proton, accents stop working instantly in Steam chat (I can only type the letter, without accent, like a, e, n) and appear like this in any in-game chat (The Elder Scrolls Online and Super Animal Royale, for example): ´a, ´e, ~a. So, all the keys are working, but the language is not being correctly recognized.

The result: when I'm playing, I can't chat with friends in Portuguese or Spanish without absurd grammatical errors. I recently stopped using Windows 10, thanks to the wonderful work you guys have done on Proton. My games run perfectly and this is the only problem, but it's annoying. I tried a lot of solution tips found across the Internet and nothing worked. Seems like this issue has been around for a while.

My system language is Brazilian Portuguese. My keyboard has the Brazilian Portuguese ABNT2 standard QWERTY layout and is working perfectly fine.

**Comment by kisak-valve** [comment]
Hello @KairanD, this sounds similar to #3641. Maybe try this workaround: https://github.com/ValveSoftware/Proton/issues/3528#issuecomment-589828273

**Comment by KairanD** [comment]
Hey, @kisak-valve , thank you very much for the fast response! I tried the workaround suggested, but, unfortunately, it didn't solve the problem. It had no effect. :(

![Captura de tela de 2020-04-23 10-10-55](https://user-images.githubusercontent.com/64203293/80105812-729ca300-854f-11ea-973f-2d7a45e2c9f3.png)

**Comment by KairanD** [comment]
Just to add more information, the same problem happens with my notebook (Asus K46CB):

Processor: Core i7 3537U @ 3.1 GHz
Memory: 8GB (2 x 4GB) @ 1600MHz.
VGA: GT 740M 2GB DDR3 128 bit.
OS: Ubuntu 18.04.4 LTS (x64) and Ubuntu 19.10 Eoan Ermine (x64). I tested both. 
VGA Driver: NVidia 390.129.
Proton Version: 5.0-6.

I also run softwares like AutoCAD 2008 (x86) and Microsoft Office 2013 (x86) through PlayOnLinux (and so Wine) and this problem doesn't happen. 

My problem can be related to #3641 and #3528 , but I'm not using the Super or Alt keys to type, since my keyboard already has accent keys natively. It can also be related to #3783 , but all my keys are working. I can type accents in game, it just doesn't detects the language correctly: for example, if I want to type "não" using the in-game chat, then the result is "n~ao". If I type in the Steam chat when a Proton game is open, the accent doesn't even show up. The strange fact is that the specific "ç" key works in both situations, so my keyboard layout is being corretly detected. 

This problem doesn't happen with Windows 10. So, since Proton runs the Windows version through Wine, the issue is definately related to a configuration in Proton. 

I'm kinda new in the Linux world, so I'm learning as I use it. My knowledge in programming is also very limited. I have been studying to learn a bit how Wine works and I have been exploring the files to see if I find a solution. If I find anything, I'll post here. I found some people online with the same problem, but not solutions.

**Comment by astroty** [comment]
> Hello @KairanD, this sounds similar to #3641. Maybe try this workaround: [#3528 (comment)](https://github.com/ValveSoftware/Proton/issues/3528#issuecomment-589828273)

I had the same issue and doing the workaround mentioned on this reply solved my issue

**Comment by KairanD** [comment]
@astroty Are you also using Ubuntu 20.04 LTS and Brazilian Portuguese language? I tried again, after reinstalling my OS, and it didn't work again. I have already accepted that I can't solve this.

**Comment by leao666** [comment]
I use Debian 11 with Cinnamon PT-BR. Steam installed from repo in Brazilian Portuguese. Proton 5.0.9

My system:
Dell Inspiron 14 7472
Core i7 8550U
Intel HD 620
NVidia MX150
16 GB Ram
ABNT2 Keyboard

I solved this problem in two games as follows:
- in the initialization options I put **PROTON_DUMP_DEBUG_COMMANDS=1 %command%** (in my game **PROTON_NO_ESYNC=1 PROTON_DUMP_DEBUG_COMMANDS=1 %command%**)
- run game and then close it
- you will have a directory like _/tmp/proton_usuario$_
- in this directory, run _./run regedit_
- go to **[HKCU\Software\Wine\]**
- create key **X11 Driver**
- create this value **UseXIM="y"** (without "")
- the result will be as in the image below

![Captura de tela de 2020-08-07 23-16-01](https://user-images.githubusercontent.com/49126762/89700376-142fe080-d904-11ea-894a-891676c642bf.png)

- you can remove the option PROTON_DUMP_DEBUG_COMMANDS=1 %command% from your game
- in the next reboot of your OS, this directory will disappear

Source: https://github.com/ValveSoftware/Proton/issues/3528#issuecomment-589828273

**Comment by KairanD** [comment]
@leao666 Thanks for the detailed info. Unfortunately, that's what I did. It's not working in Ubuntu 20.04.1 LTS or 20.10. It's an extremely annoying problem... And I have already tried everything.

**Comment by KairanD** [comment]
Now I've already tried (with the solution above):
    1. Different versions of Proton (4.11-13, 5.0-9 and 5.13-1);
    2. Disabling all the GNOME Extensions;
    3. Starting with `LC_ALL=pt_BR.UTF-8 LANG=pt_BR.UTF-8 %command%`;
    4. A combination of everything above.

Nothing worked:
![image](https://user-images.githubusercontent.com/64203293/97294818-40853a00-182d-11eb-9aa9-d087ba5efcdc.png)

**Comment by KairanD** [comment]
**GOT IT WORKING!** With a fresh install of Ubuntu 20.10. The problem was my system. Now I can confirm that the solution above works. In my case, it wasn't necessary to run with `LC_ALL=pt_BR.UTF-8 LANG=pt_BR.UTF-8 %command%`.

**Comment by KairanD** [comment]
There is an easier solution: use the option `XMODIFIERS="" %command%` for each affected game.

Also, this problem does not happen on Arch Linux. Only on Debian, Ubuntu and Ubuntu-based systems. I don't know why.

**Comment by malcmerlyn** [comment]
> There is an easier solution: use the option `XMODIFIERS="" %command%` for each affected game.
> 
> Also, this problem does not happen on Arch Linux. Only on Debian, Ubuntu and Ubuntu-based systems. I don't know why.

Sorry but not all debian and Ubuntu based distro. I had the exact same issue on Ubuntu 24.04 with a french keyboard and playing Guild Wars 2. The incident cannot be reproduced when I replaced Ubuntu 24.04 with Xubuntu 24.04 and now Linux Mint 21.3, the game recognize the accented characters in these two Linux fresh installs. For me the issue is Ubuntu related

**Comment by KairanD** [comment]
This issue still persists in Debian 13. The workarounds still work, but a definitive solution is needed.

**Comment by JohnnyDoesStuff** [comment]
> There is an easier solution: use the option `XMODIFIERS="" %command%` for each affected game.
> 
> Also, this problem does not happen on Arch Linux. Only on Debian, Ubuntu and Ubuntu-based systems. I don't know why.

Adding the part `XMODIFIERS="" %command%` did some improvement for me but something is still wrong. In my case it's about german keys (ä, ö, ü, ...) and they kinda seem to work. In chat they are properly displayed and i can use accents as well like á. My issue is that the keys also seem to be mapped to something else:
- ä is kinda understood as F13 (which i don't even got on my keyboard physically)
- ö becomes some kind of ESC as well
- ü becomes some kind of A-key + Enter as well

So i guess that some keys are understood as keys of a Controller. Well, i haven't connected a controller. Does anyone have an idea what i might be doing wrong?

I'm on Ubuntu 24.04

**Comment by m2scared** [comment]
tldr

```bash
# Set wine IME settings
#
# Allows '~ + a = ã' when IME is enable
export XMODIFIERS=@im=ibus
export GTK_IM_MODULE=ibus
export QT_IM_MODULE=ibus
ibus-daemon -drx

# Debug
# export WINEDEBUG=+keyboard

exec /usr/bin/wine notepad
```

I have had a related issue when using **Kasm Desktop** with `wine`. By following the steps in that thread and making some adjustments, it worked. But I'm not sure how exactly it fixed the problem.

https://forum.literatureandlatte.com/t/accented-characters-dont-work-on-wine-linux/34644/16

What I notice is that **kasm Desktop** gives:

```
 2025-07-01 12:14:18,867 [INFO] Input: Added unknown keysym 0xe3 to keycode 222
```

And `wine`gets:

```
0124:trace:keyboard:KEYBOARD_MapDeadKeysym no character for dead keysym 0x0000fe03
```

I hope it can help somehow.
